test/Microsoft.Dnx.Compilation.CSharp.Tests/RoslynCompilerTest.cs

```diff
@@ -79,6 +81,51 @@ public void FlowsProjectPropertiesIntoAssembly()
             Assert.All(compilationAttributes, compilationAttribute => expectedAttributes[compilationAttribute.AttributeClass.ToString()].Equals(
                 compilationAttribute.ConstructorArguments.First().Value));
         }
-#endif
+
+        [Fact]
+        public void DoesNotSignAndEmitEntyPoinInPreprocessAssembly()
+        {
+            const string testFrameworkName = "DNX,Version=v4.5.1";
+
+            // Arrange
+            var compilerOptions = new FakeCompilerOptions()
+            {
+                KeyFile = "keyfile.snk",
+                DelaySign = true,
+                EmitEntryPoint = true
+            };
+
+            var cacheContextAccessor = new FakeCacheContextAccessor { Current = new CacheContext(null, (d) => { }) };
+
+            var compilationProjectContext = new CompilationProjectContext(
+                new CompilationTarget("Test", new FrameworkName(testFrameworkName), string.Empty, "preprocess"),
+                Directory.GetCurrentDirectory(),
+                "project.json",
+                string.Empty,
+                string.Empty,
+                string.Empty,
+                string.Empty,
+                new Version(0,0,0,0),
+                false,
+                new CompilationFiles(
+                    new List<string> { },
+                    new List<string> { }),
+                compilerOptions);
+            var compiler = new RoslynCompiler(null, cacheContextAccessor, new FakeNamedDependencyProvider(), null, null, null);
+
+            var metadataReference = new FakeMetadataReference() { MetadataReference = MetadataReference.CreateFromFile(typeof(object).Assembly.Location) };
+
+            // Act
+            var compilationContext = compiler.CompileProject(
+                compilationProjectContext,
+                new List<IMetadataReference> { metadataReference },
+                new List<ISourceReference> { },
+                () => new List<ResourceDescriptor>());
+
+            Assert.Equal(OutputKind.DynamicallyLinkedLibrary, compilationContext.Compilation.Options.OutputKind);
+            Assert.Equal(null, compilationContext.Compilation.Options.CryptoKeyFile);
+            Assert.Equal(0, compilationContext.Compilation.Options.CryptoPublicKey.Length);
```

Comment:
Do you need a separate test for this? You did not set `useOssSigning` so you can't tell if it was reset or not. On the other hand I don't think you can test `keyFile` and `useOssSigning:true` in one test because `Options.CryptoKeyFile will never be set.`